dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -3054,4 +3055,31 @@ public void deleteOtherRelation(Project project, Map<String, Object> result, Pro
 
     }
 
+    /**
+     * batch release process definitions based on codes and releaseState.
+     *
+     * @param loginUser    login user
+     * @param projectCode  project code
+     * @param codes        process definition code list
+     * @param releaseState releaseState
+     * @return Result of the batch release process definitions.
+     */
+    @Override
+    @Transactional
+    public Map<String, Object> batchReleaseProcessDefinitionsByCodes(User loginUser, long projectCode, String codes,
+                                                                     ReleaseState releaseState) {
+        Map<String, Object> result = new HashMap<>();
+        if (StringUtils.isEmpty(codes)) {
+            log.error("Parameter processDefinitionCodes is empty, projectCode is {}.", projectCode);
```

Comment:
how about use info or warn as log level?

Reply:
Yes, I feel it's better to use warn. I'll make some changes then